metricbeat/module/rabbitmq/node/data.go

```diff
@@ -129,27 +128,22 @@ var (
 	}
 )
 
-func eventsMapping(content []byte) ([]common.MapStr, error) {
+func eventsMapping(r mb.ReporterV2, content []byte) {
 	var nodes []map[string]interface{}
 	err := json.Unmarshal(content, &nodes)
 	if err != nil {
-		logp.Err("Error: ", err)
-		return nil, err
+		r.Error(err)
+		return
 	}
 
-	events := []common.MapStr{}
-	errors := s.NewErrors()
-
 	for _, node := range nodes {
-		event, errs := eventMapping(node)
-		events = append(events, event)
-		errors.AddErrors(errs)
-
+		eventMapping(r, node)
 	}
-
-	return events, errors
 }
 
-func eventMapping(node map[string]interface{}) (common.MapStr, *s.Errors) {
-	return schema.Apply(node)
+func eventMapping(r mb.ReporterV2, node map[string]interface{}) {
+	event, _ := schema.Apply(node)
```

Comment:
I tried not to ignore errors here, but `Apply` is returning an error that seems like an empty list of missing fields (Literally: "`Required fields are missing: `"). This happens also with the existing code, so this is not a regression. I leave investigating this for a future PR.

Reply:
Yes, we need a better general approach here. LGTM for now. Check out my recent changes in the elasticsearch module. There I use the errors for testing purpose only.